src/test/codegen/repr-transparent-x86_64.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// only-x86_64
+
+// compile-flags: -C no-prepopulate-passes
+//
+#![crate_type="lib"]
+#![feature(repr_transparent)]
+
+#[repr(C)]
+pub struct Rgb8 { r: u8, g: u8, b: u8 }
+
+#[repr(transparent)]
+pub struct Rgb8Wrap(Rgb8);
+
+// NB: closing parenthesis is missing because sometimes the argument has a name and sometimes not
```

Comment:
Aside: I think this was an artifact of the previous hack and shouldn't be necessary any more. Not that important, though.

Reply:
That shouldn't have made a difference. The only thing that should make a difference here is whether e.g. `--emit llvm-ir` is given, making LLVM preserve names.

Reply:
I'm pretty sure the difference was due to declaring the function as taking an aggregate vs taking a primitive, and possibly also platform dependent. But whatever, it should definitely be consistent now.

Reply:
Oh, you're right, I botched my local test and passed the wrong struct.